### PR TITLE
[master] Disable EVM functionality according to flag in config

### PR DESCRIPTION
**Pull request by CSideSteve**
## Description

Disable RPC methods for eth when EVM is disabled.

## Backward Compatibility

- [ ] This is not a breaking change
- [ ] This is a breaking change

### Implementation

- [ ] **ready for review**

### Integration Test (Core Team)

- [ ] local machine test

- [ ] small-scale cloud test

### CODE REVIEW

File: src/libServer/IsolatedServer.cpp [hunk 1]
```diff
@@ -126,96 +126,98 @@ IsolatedServer::IsolatedServer(Mediator& mediator,
...
-                         jsonrpc::JSON_STRING, "param01", jsonrpc::JSON_OBJECT,
-                         NULL),
-      &LookupServer::GetEthCallZilI);
-
-  AbstractServer<IsolatedServer>::bindAndAddMethod(
-      jsonrpc::Procedure("eth_call", jsonrpc::PARAMS_BY_POSITION,
-                         jsonrpc::JSON_STRING, "param01", jsonrpc::JSON_OBJECT,
-                         "param02", jsonrpc::JSON_STRING, NULL),
-      &LookupServer::GetEthCallEthI);
+  if (ENABLE_EVM) {
```
[the remark this answers is not in the document]
Comment (CSideSteve) [reply src/libServer/IsolatedServer.cpp:129]: You could raise a new ticket for this although it is less critical as isolatedServer is a test and development tool and does not run on the mainnet.

The purpose of this switch is to make sure that the first release does not expose EVM functionality on Mainnet, if there is that possibility then the code cannot be merged onto the mainnet release for November.

[the remark this answers is not in the document]
Comment (CSideSteve) [reply src/libServer/IsolatedServer.cpp:129]: Also see changes made to LookupServer which is on mainnet with the EthRcpMethods plug in that does exactly this for the RPC calls.

[the remark this answers is not in the document]
Comment (CSideSteve) [reply src/libServer/IsolatedServer.cpp:165]: Resolved in latest commit

File: src/libServer/IsolatedServer.cpp [hunk 2]
```diff
@@ -248,114 +158,214 @@ IsolatedServer::IsolatedServer(Mediator& mediator,
...
   }
+  BindAlllEvmMethods();
+}
 
-  // Add the JSON-RPC eth style methods
-  AbstractServer<IsolatedServer>::bindAndAddMethod(
-      jsonrpc::Procedure("eth_blockNumber", jsonrpc::PARAMS_BY_POSITION,
-                         jsonrpc::JSON_STRING, NULL),
-      &IsolatedServer::GetEthBlockNumberI);
+void IsolatedServer::BindAlllEvmMethods() {
```
Comment (moboware) [inline src/libServer/IsolatedServer.cpp:164]: Alll with 2 l's

**Review by moboware — COMMENTED** [review]
one typo

File: src/libServer/IsolatedServer.cpp (hunk 2 above)
Comment (CSideSteve) [reply src/libServer/IsolatedServer.cpp]: Corrected

File: constants.xml [hunk 3]
```diff
@@ -191,7 +191,7 @@
         <!-- Timeout in seconds for ONLY connection that reach our callback function, 0 means no timeout-->
         <CONNECTION_CALLBACK_TIMEOUT>0</CONNECTION_CALLBACK_TIMEOUT>
         <ENABLE_EVM>true</ENABLE_EVM>
-        <EVM_SERVER_BINARY>/usr/local/bin/evm-ds</EVM_SERVER_BINARY>
+        <EVM_SERVER_BINARY>/home/stephen/versions/Zilliqa/target/release/evm-ds</EVM_SERVER_BINARY>
```
Comment (moboware) [inline constants.xml:194]: remove your name from here

File: constants.xml [hunk 4]
```diff
@@ -288,7 +288,7 @@
     </recovery>
     <smart_contract>
         <ENABLE_SC>true</ENABLE_SC>
-        <SCILLA_ROOT>/scilla</SCILLA_ROOT>
+        <SCILLA_ROOT>/home/stephen/run/scilla</SCILLA_ROOT>
```
Comment (moboware) [inline constants.xml:291]: and here too

Comment (CSideSteve) [reply constants.xml:291]: This should not be committed, good catch, no changes to constants.xml required. We are looking at changing the way we use it locally so you don't need to keep editing.

**Review by moboware — APPROVED** [review]

**Pull request merged**